throttle/throttledRecorder_test.go

```diff
@@ -160,7 +171,7 @@ func TestCanHaveNoSparseRecordings(t *testing.T) {
 		SparseAfter:     11,
 		SparseLength:    0,
 	}
-	recorder := NewThrottledRecorder(baseRecorder, config, 1)
+	recorder := NewThrottledRecorder(baseRecorder, new(ThrottledCounter), config, 1)
```

Comment:
why are you passing a listener in these tests if it's not being used? just pass nil if it's not relevant to the test. 
(also below)